pkg/controller/atlasdeployment/serverless_deployment.go

```diff
@@ -54,3 +77,15 @@ func ensureServerlessInstanceState(ctx *workflow.Context, project *mdbv1.AtlasPr
 		return atlasDeployment, workflow.Terminate(workflow.Internal, fmt.Sprintf("unknown deployment state %q", atlasDeployment.StateName))
 	}
 }
+
+func isTagsEqual(a []*mongodbatlas.Tag, c []*mongodbatlas.Tag) bool {
+	if len(a) == len(c) {
+		for i, aTags := range a {
+			if aTags.Key != (c)[i].Key || aTags.Value != (c)[i].Value {
+				return false
+			}
+		}
+		return true
+	}
+	return false
+}
```

Comment:
I see we decided on making the order matter. This is fine, except we now need to discard the following pathologic case: Say we have the pair `{A=a,B=b}` in Atlas, but the k8s spec says `{B=b,A=a}`

1. This detects a change, because the order does not match.
2. A reconcile does patch the tags, but as the tags are present, Atlas makes no real changes on its side, **and keeps the old order**.
3. Next reconcile, we start again from 1 and we are in a loop reconciling unordered tags that do not need to be reconciled.

Do we have any test confirming or discarding that this can happen with Atlas tags?

Reply:
Thanks for double checking. No more objections then.